src/Stack/Ghci.hs

```diff
@@ -192,7 +192,7 @@ ghciSetup mainIs stringTargets = do
         forM locals $
         \(name,(cabalfp,components)) ->
              makeGhciPkgInfo sourceMap (map fst locals) name cabalfp components
-    unless (M.null realTargets) (build (const (return ())) Nothing bopts)
+    try (unless (M.null realTargets) (build (const (return ())) Nothing bopts)) >>= (\x -> return (x :: Either SomeException ()))
```

Comment:
This is not great, both in that it needs the extra `>>=` to do the type annotation and it listens to all exception types. For the type annotation, I tried annotated with `try (build ...) :: m (Either SomeException ())` but ghc can't resolve `m` with `m` used in the type of the function. I tried to use `ScopedTypeVariables` but couldn't make it do what I want. Any ideas?